src/openlifu/util/volume_conversion.py

```diff
@@ -0,0 +1,157 @@
+"""Utilities for converting between different medical imaging formats."""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import nibabel as nib
+import numpy as np
+import pydicom
+
+from openlifu.util.types import PathLike
+
+
+def is_dicom_file_or_directory(path: PathLike) -> bool:
+    """
+    Check if a path is a DICOM file or directory containing DICOM files.
+
+    Args:
+        path: Path to check
+
+    Returns:
+        True if path is a DICOM file or directory with DICOM files, False otherwise
+    """
+    path = Path(path)
+
+    if path.is_file():
+        # check for 'DICM' magic bytes at offset 128
+        try:
+            with open(path, 'rb') as f:
+                f.seek(128)
+                return f.read(4) == b'DICM'
+        except OSError:
+            return False
+
+    elif path.is_dir():
+        for file in path.iterdir():
+            if file.is_file():
+                try:
+                    with open(file, 'rb') as f:
+                        f.seek(128)
+                        if f.read(4) == b'DICM':
+                            return True
+                except OSError:
+                    continue
+
+    return False
+
+
+def extract_affine_from_dicom(dicom_slices: list) -> np.ndarray:
+    """
+    Extract the affine transformation matrix from DICOM header information.
+    Converts from DICOM LPS (Left-Posterior-Superior) to NIfTI RAS.
+
+    Args:
+        dicom_slices: List of tuples (instance_number, pixel_array, dicom_dataset)
```

Comment:
Can you elaborate a little on what `dicom_dataset` means here? Is it a metadata format from pydicom?